src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java

```diff
@@ -119,7 +119,6 @@ public void visitToken(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         if (parent.getType() == TokenTypes.SLIST
                 && (!allowInSwitchCase
-                    || parent.getParent().getType() != TokenTypes.CASE_GROUP
```

Comment:
I am unsure why this isn't needed and I couldn't find a case through regression.